cpp/src/arrow/csv/options.h

```diff
@@ -119,15 +119,20 @@ struct ARROW_EXPORT ReadOptions {
 
   /// Whether to use the global CPU thread pool
   bool use_threads = true;
-  /// Block size we request from the IO layer; also determines the size of
-  /// chunks when use_threads is true
+
+  /// \brief Block size we request from the IO layer.
+  ///
+  /// This will determine multi-threading granularity as well as
```

Comment:
I'm not sure how multi-threading is related to block size at the moment.  In the future we could add parallel readahead but currently both the file and streaming CSV readers do not do this.  The threaded CSV reader is threaded across the block and not multiple blocks at once.

Reply:
The CSV reader processes each block separately, and therefore potentially on a different thread. Additionally, each column in each block is converted separately as well.